pkg/healthcheck/healthcheck.go

```diff
@@ -701,6 +714,122 @@ func (hc *HealthChecker) allCategories() []category {
+					hintAnchor:  "cni-plugin-cm-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						_, err := hc.kubeAPI.CoreV1().ConfigMaps(hc.ControlPlaneNamespace).Get(cniConfigMapName, metav1.GetOptions{})
+						if err != nil {
+							return err
+						}
+						return nil
+					},
+				},
+				{
+					description: "cni plugin PodSecurityPolicy exists",
+					hintAnchor:  "cni-plugin-psp-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkPodSecurityPolicies(true, []string{fmt.Sprintf("linkerd-%s-cni", hc.ControlPlaneNamespace)}, false)
+					},
+				},
+				{
+					description: "cni plugin ClusterRole exists",
+					hintAnchor:  "cni-plugin-cr-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkClusterRoles(true, hc.expectedCniRBACNames(), false)
+					},
+				},
+				{
+					description: "cni plugin ClusterRoleBinding exists",
+					hintAnchor:  "cni-plugin-crb-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkClusterRoleBindings(true, hc.expectedCniRBACNames(), false)
+					},
+				},
+				{
+					description: "cni plugin Role exists",
+					hintAnchor:  "cni-plugin-r-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkRoles(true, hc.expectedCniRBACNames(), false)
+					},
+				},
+				{
+					description: "cni plugin RoleBinding exists",
+					hintAnchor:  "cni-plugin-rb-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkRoleBindings(true, hc.expectedCniRBACNames(), false)
+					},
+				},
+				{
+					description: "cni plugin ServiceAccount exists",
+					hintAnchor:  "cni-plugin-sa-exists",
+					fatal:       true,
+					check: func(context.Context) error {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						return hc.checkServiceAccounts(expectedCniServiceAccountNames)
+					},
+				},
+				{
+					description: "cni plugin DaemonSet exists",
+					hintAnchor:  "cni-plugin-ds-exists",
+					fatal:       true,
+					check: func(context.Context) (err error) {
+						if !hc.NoInitContainer {
+							return &SkipError{Reason: cniDisabledSkipReason}
+						}
+						hc.cniDaemonSet, err = hc.kubeAPI.GetDaemonSet(hc.ControlPlaneNamespace, expectedCniDsName)
```

Comment:
why do we create a dedicated method for getting the DS instead of just doing `hc.kubeAPI.CoreV1()...` directly like we do for configmaps in an above check.